Lundalogik/lime-elements
# Deprecate `iconColor`, `iconBackgroundColor` etc, in favour of using the `Icon` interface

fix https://github.com/Lundalogik/lime-elements/issues/2505

💡 Don't get scared of the number of files changed in this PR. Most of the files are the updated examples that now use the new `Icon` interface. The logic of deprecating the old props works exactly the same for all components. Once you see and understand one of them, you will have an easy time understanding the rest of them.

### Which components are updated:
This is the list of component that are affected:

- [x] action bar
- [x] chipset
- [x] progress flow
- [x] picker
- [x] tab bar
- [x] file





# How to test this PR?

`iconColor`, `iconFillColor`, `iconBackgroundColor` and `iconTitle` props are deprecated now in different components. But they should still work if they are used.

The old way may look like this:
```tsx
icon: 'name',
iconFillColor: 'black',
iconBackgroundColor: 'white',
iconTitle: 'title',
```
And the new way looks like this:
```tsx
icon: {
	name: 'name',
    color: 'black',
    backgroundColor: 'white',
    title: 'title',
},
```

⚠️ Keep in mind that the old way still work, but the combination of the old and new will not work! So for example a config like below wouldn't work and the `iconColor` would not get applied:
```tsx
icon: {
	name: 'name',
},
iconColor: 'color'
```

An easy test is to link this to the web client (where all the configs in crm-components are still old) and see if each component still show the colors as expected. 

Otherwise, a manual test is to change back the "examples" in Lime Elements docs to what they were before and make sure that each component renders the colors as expected.


## Review:

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:

Linux:

macOS:

Mobile:
